util/src/main/java/tc/oc/pgm/util/xml/XMLUtils.java

```diff
@@ -400,50 +403,108 @@ public static <T extends Number & Comparable<T>> Range<T> parseNumericRange(
    *
    * <p>for a closed-open range from 0 to 1
    *
+   * <p>Or in the vanilla Minecraft format e.g.
+   *
+   * <p>1..5 or ..5 or 1..
+   *
+   * <p>equal to [1, 5], (-oo, 5] and [1, oo)
+   *
    * <p>Also supports singleton ranges derived from providing a number with no delimiter
+   *
+   * @implNote Since infinity and "infinity"({@link Double#POSITIVE_INFINITY} etc.) is handled
+   *     differently by the Google ranges we find the infinities and create Ranges using {@link
+   *     Range#upTo(Comparable, BoundType) Range.upTo} and {@link Range#downTo(Comparable,
+   *     BoundType) Range.downTo} instead of resolving to the max or min value of the range type.
+   *     (Like {@link #parseNumber(String, Class, boolean)} does)
    */
   public static <T extends Number & Comparable<T>> Range<T> parseNumericRange(
-      @NotNull Node node, Class<T> type) throws InvalidXMLException {
-    Matcher matcher = RANGE_RE.matcher(node.getValue());
-    if (!matcher.matches()) {
-      T value = parseNumber(node, node.getValue(), type, true);
+      Node node, Class<T> type) throws InvalidXMLException {
+    String nodeValue = node.getValue();
+
+    String lower;
+    BoundType lowerBound = null;
+    String higher;
+    BoundType higherBound = null;
+
+    Matcher matcher = RANGE_RE.matcher(nodeValue);
+
+    Matcher legacy = RANGE_LEGACY.matcher(nodeValue);
+
+    if (legacy.matches()) {
+      String first = legacy.group(1);
+      String second = legacy.group(2);
+
+      if (first == null) {
+        higher = second.trim();
+        higherBound = BoundType.CLOSED;
+        lower = "-oo";
+      } else if (second == null) {
+        lower = first.trim();
+        lowerBound = BoundType.CLOSED;
+        higher = "oo";
+      } else {
+        lower = first.trim();
+        lowerBound = BoundType.CLOSED;
+        higher = second.trim();
+        higherBound = BoundType.CLOSED;
+      }
```

Comment:
```suggestion
      lower = first == null ? "-oo" : first.trim();
      upper = second == null ? "oo" : second.trim();
      lowerBound = BoundType.CLOSED;
      higherBound = BoundType.CLOSED;
```
Currently you're assuming that just because first is null, second won't be, both could be null if you just type `..` for an unbounded range of all numbers. Additionally this simplifies the logic